### PR TITLE
VisIt: Update to VisIt 3.3.1.

### PR DESCRIPTION
**Pull request by brugger1**
Updated the version of VisIt to 3.3.1.

I tested this on crusher.olcf.ornl.gov using the following install line.
```
spack install visit%gcc@11.2.0+mpi+gui ^python@3.7.7+shared ^mesa@21.2.5+opengl ^llvm@11.0.1 ^vtk@8.1.0
```

### CODE REVIEW

**Review by cyrush — DISMISSED** [review]

**Review by tldahlgren — DISMISSED** [review]
Confirmed sha256.

**Comment by brugger1** [comment]
I see the e4s_power_pr_build test failed. The other e4s tests all passed and I noticed that the e4s_power_pr_build test timed out at 6 hours. Is this just a timeout issue? Is it documented somewhere how to investigate this further, since the log file was truncated.

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by brugger1** [comment]
@alalazo , @tldahlgren the `e4s-power-pr-build` CI test timed out again after 360 minutes. I don't know if there is a real problem or if the test ran out of time. The log file has been truncated because of the size limit and everything looks fine up to that point. I don't have a power system to try and debug this with the e4s-power container. How do I proceed?

**Review by alalazo — DISMISSED** [review]

**Review by tldahlgren — DISMISSED** [review]

**Comment by brugger1** [comment]
The CI now shows 3 completely new failures that I can't see how my changes could be causing.

image

For data-vis-sdk-pr-generate I get:
```
$ export SPACK_CI_CONFIG_ROOT="${SPACK_ROOT}/share/spack/gitlab/cloud_pipelines/configs"
$ spack --config-scope "${SPACK_CI_CONFIG_ROOT}" --config-scope "${SPACK_CI_CONFIG_ROOT}/${SPACK_TARGET_PLATFORM}" --config-scope "${SPACK_CI_CONFIG_ROOT}/${SPACK_TARGET_PLATFORM}/${SPACK_TARGET_ARCH}" ${CI_STACK_CONFIG_SCOPES} ci generate --check-index-only --buildcache-destination "${SPACK_BUILDCACHE_DESTINATION}" --artifacts-root "${CI_PROJECT_DIR}/jobs_scratch_dir" --output-file "${CI_PROJECT_DIR}/jobs_scratch_dir/cloud-ci-pipeline.yml"
==> Starting concretization pool with 6 processes
==> Error: errors occurred during concretization of the environment:
ecp-data-vis-sdk '~visit' conflicts with '^visit'
ecp-data-vis-sdk '~visit' conflicts with '^visit'
ecp-data-vis-sdk '~visit' conflicts with '^visit'
Running after_script
```
which I don't get when I try it manually on an x86_64 system with the latest develop.

For e4s-power-pr-generate and e4s-pr-generate I get:
```
$ spack --config-scope "${SPACK_CI_CONFIG_ROOT}" --config-scope "${SPACK_CI_CONFIG_ROOT}/${SPACK_TARGET_PLATFORM}" --config-scope "${SPACK_CI_CONFIG_ROOT}/${SPACK_TARGET_PLATFORM}/${SPACK_TARGET_ARCH}" ${CI_STACK_CONFIG_SCOPES} ci generate --check-index-only --buildcache-destination "${SPACK_BUILDCACHE_DESTINATION}" --artifacts-root "${CI_PROJECT_DIR}/jobs_scratch_dir" --output-file "${CI_PROJECT_DIR}/jobs_scratch_dir/cloud-ci-pipeline.yml"
==> Starting concretization pool with 16 processes
==> Error: No version for 'hdf5' satisfies '@1.14.0' and '@1.8.21'
Running after_script
```
which I don't really understand.

I did a merge of the latest develop into the branch and I still got these failures.

I was getting a failure with silo 4.11 building with gcc 11.1, so I added the constraint on it to 4.10. This also suddenly appeared. When I added that constraint I started getting the errors above.

**Comment by tldahlgren** [comment]
@alalazo Do the concretizer failures make sense to you?

File: var/spack/repos/builtin/packages/visit/package.py [hunk 1]
```diff
@@ -137,7 +138,8 @@ class Visit(CMakePackage):
 
     # VisIt uses Silo's 'ghost zone' data structures, which are only available
     # in v4.10+ releases: https://wci.llnl.gov/simulation/computer-codes/silo/releases/release-notes-4.10
-    depends_on("silo@4.10: +shared", when="+silo")
+    # Silo 4.11 doesn't compile with gcc 11.1.
```
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/visit/package.py:141]: If this is a GCC conflict for Silo, then it should probably be expressed as a `conflicts("%gcc@11.1:", when="@4.11")` in the Silo package, not in the VisIt package. This way you can still build VisIt wth silo 4.11 when using compilers that do work.

**Review by kwryankrattiger — CHANGES_REQUESTED** [review]
Move silo related constraints so `silo/package.py`

**Comment by kwryankrattiger** [comment]
I have been seeing these failures when trying to use a newer HDF5. 

See here #35195

File: var/spack/repos/builtin/packages/visit/package.py (hunk 1 above)
Comment (brugger1) [reply var/spack/repos/builtin/packages/visit/package.py:141]: @kwryankrattiger got it. I'll submit a PR for that.

**Pull request closed without merging**